docs/source/fx.rst

```diff
@@ -293,10 +293,188 @@ Writing Transformations
+traditional debugging techniques like ``print`` statements or ``pdb`` is
+ not as straightfoward. Luckily, we have several techniques we can use
+for debugging the generated code.
+
+``pdb``
+~~~~~~~~~~~~
+Use ``pdb`` to step into the running program. Although the code that
+represents the FX graph is not in any source file, we can still step
+into it manually using ``pdb`` when the forward pass is invoked.
+
+::
+    x, y = torch.rand(2, 3), torch.rand(2, 3)
+    m = symbolic_trace(M())
+    import pdb; pdb.set_trace()
+    m(x, y)
+
+Print the generated code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you’d like to run the same code multiple times, then it can be
+a bit tedious to step to the right code with ``pdb``. In that case, one
+approach is to simply copy-paste the generated ``forward`` pass into
+your code and examine it from there.
+
+::
+
+    # Module whose trace we want to debug
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.x = torch.nn.Parameter(torch.rand(4, 3))
+
+        def forward(self, y):
+            return self.x + y
+
+    # Symbolically trace an instance of `M`
+    traced = symbolic_trace(M())
+
+    # Print the code generated from symbolic tracing. This outputs:
+    #   def forward(self, y):
+    #       x = self.x
+    #       add_1 = x + y;  x = y = None
+    #       return add_1
+    # Copy this code for later
+    print(traced)
+
+    # Subclass the original Module
+    class SubclassM(M):
+        def __init__(self):
+            super().__init__()
+
+        # Paste the generated `forward` function (the one we printed and
+        # copied on line 22) here
+        def forward(self, y):
+            x = self.x
+            add_1 = x + y;  x = y = None
+            return add_1
+
+    # Create an instance of the original, untraced Module. Then, create an
+    # instance of the Module with the copied `forward` function. We can
+    # now compare the output of both the original and the traced version.
+    pre_trace = M()
+    post_trace = SubclassM()
+
+
+Use the ``to_folder`` function from ``GraphModule``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``to_folder`` is a method in ``GraphModule`` that allows you to dump
+out the generated FX code to a folder. Although copying the forward pass
+into the code often suffices as in :ref:`Print the generated code
+<the above section>`, it doesn’t capture any model attribute state.
+To examine modules and parameters, we can use ``to_folder``.
+
+::
+    m = symbolic_trace(M())
+    m.to_folder("foo", "Bar")
+    from foo import Bar
+    y = Bar()
+
+After running the above example, we can then look at the code within
+``foo/module.py`` and modify it as desired to debug the generated code.
+
+Debugging the Transformation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, read the `FX
+documentation <https://pytorch.org/docs/master/fx.html>`__. Understand
```

Comment:
This is linking to the page the user is on :). I think it would suffice to `:ref:` to the `Node`, `Graph`, `Proxy` and `Tracer` sections here